From 05f656c1846cb36c95ada40667a153e64d796909 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Tue, 12 Apr 2016 18:57:29 +0800
Subject: [PATCH] ARM64: dts: gru: Add support machine driver for DA7219

Now, we can playback and capture via DA7219 machine driver call the
da7219_aad_jack_det (simple-card can not do this).

Change-Id: I8b1be189031f875b1c5328e9357115761a5f4da3
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi | 49 ++++++++++++--------
 1 file changed, 30 insertions(+), 19 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi
index b94426128199..234bd79c8e3c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi
@@ -346,6 +346,9 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&pp1800_audio_en>;
 
+		regulator-always-on;
+		regulator-boot-on;
+
 		enable-active-high;
 		gpio = <&gpio0 1 GPIO_ACTIVE_HIGH>;
 
@@ -470,6 +473,13 @@
 
 		pmu1830-supply = <&pp1800_pmu>;		/* PMUIO2_VDD */
 	};
+
+	sound {
+		compatible = "rockchip,rockchip-audio-da7219";
+		rockchip,model = "audio-da7219";
+		rockchip,i2s-controller = <&i2s0>;
+		rockchip,audio-codec = <&codec>;
+	};
 };
 
 &cpu_l0 {
@@ -569,6 +579,7 @@ ap_i2c_audio: &i2c8 {
 	codec: da7219@1a {
 		compatible = "dlg,da7219";
 		reg = <0x1a>;
+
 		pinctrl-names = "default";
 		pinctrl-0 = <&headset_int_l>;
 
@@ -584,25 +595,25 @@ ap_i2c_audio: &i2c8 {
 		clock-names = "mclk";
 
 // Below is from bindings; we need to adjust for our SoC / board.
-// 		dlg,ldo-lvl = <1200>;
-// 		dlg,micbias-lvl = <2600>;
-// 		dlg,mic-amp-in-sel = "diff";
-//
-// 		da7219_aad {
-// 			dlg,btn-cfg = <50>;
-// 			dlg,mic-det-thr = <500>;
-// 			dlg,jack-ins-deb = <20>;
-// 			dlg,jack-det-rate = "32ms_64ms";
-// 			dlg,jack-rem-deb = <1>;
-//
-// 			dlg,a-d-btn-thr = <0xa>;
-// 			dlg,d-b-btn-thr = <0x16>;
-// 			dlg,b-c-btn-thr = <0x21>;
-// 			dlg,c-mic-btn-thr = <0x3E>;
-//
-// 			dlg,btn-avg = <4>;
-// 			dlg,adc-1bit-rpt = <1>;
-// 		};
+		dlg,ldo-lvl = <1200>;
+		dlg,micbias-lvl = <2600>;
+		dlg,mic-amp-in-sel = "diff";
+
+		da7219_aad {
+			dlg,btn-cfg = <50>;
+			dlg,mic-det-thr = <500>;
+			dlg,jack-ins-deb = <20>;
+			dlg,jack-det-rate = "32ms_64ms";
+			dlg,jack-rem-deb = <1>;
+
+			dlg,a-d-btn-thr = <0xa>;
+			dlg,d-b-btn-thr = <0x16>;
+			dlg,b-c-btn-thr = <0x21>;
+			dlg,c-mic-btn-thr = <0x3E>;
+
+			dlg,btn-avg = <4>;
+			dlg,adc-1bit-rpt = <1>;
+		};
 	};
 };
 
-- 
2.35.3

